examples/state_management_example/README.md

```diff
@@ -0,0 +1,160 @@
+
+## Overview
+
+The state management system provides:
+
+- Support for multiple state backends (Observable/Redux)
+- Type-safe state management
+- Efficient component re-rendering
+- Consistent API through hooks
+
+## Core Components
+
+### StateAdapter
+
+The adapter pattern provides a unified interface for different state backends:
+
+```typescript
+// For RxJS Observables
+const observableAdapter = new BehaviorSubjectAdapter(subject);
+
+// For Redux stores
+const reduxAdapter = new ReduxStateAdapter(store, state => state.mySlice);
+```
+
+### BaseActions
+
+Define state-changing logic that works with any state backend:
+
+```typescript
+class CounterActions extends BaseActions<CounterState> {
+  // Works with both Observable and Redux backends
+  constructor(adapterOrObservable: StateAdapter<CounterState> | ObservableLike<CounterState>) {
+    super(adapterOrObservable);
+  }
+
+  increment() {
+    this.updateState(state => ({
+      ...state,
+      count: state.count + 1
+    }));
+  }
+}
+```
+
+### BaseSelectors
+
+Access state in a read-only manner:
+
+```typescript
+class CounterSelectors extends BaseSelectors<CounterState> {
+  // Works with both Observable and Redux backends
+  constructor(adapterOrObservable: StateAdapter<CounterState> | ObservableLike<CounterState>) {
+    super(adapterOrObservable);
+  }
+  
+  // The base class provides getState() by default
+  isPositive() {
+    return this.getState().count > 0;
+  }
+}
+```
+
+## React Hooks
+
+### useAction
+
+Access actions from a registered service:
+
+```typescript
+// In a React component
+const actions = useAction<CounterState>('counterService');
+// Then use actions
+<button onClick={() => actions.increment()}>+</button>
+```
+
+### useSelector
+
+Subscribe to state changes:
+
+```typescript
+// Only re-renders when the selected value changes
+const count = useSelector<CounterState, number>(
+  'counterService',
+  state => state.count
+);
+```
+
+## Example Implementations
+
+### Observable-Based Counter
+
+```typescript
+// Create with Observable pattern
+const state$ = new BehaviorSubject<CounterState>({ count: 0 });
+const actions = new CounterActions(state$);
+const selectors = new CounterSelectors(state$);
+
+// Register with global store
+globalStoreServiceRegister('observableCounter', selectors, actions);
+```
+
+### Redux-Based Counter
+
+```typescript
+// Create with Redux pattern
+const store = configureStore({
+  reducer: { counter: counterReducer }
+});
+const adapter = new ReduxStateAdapter<CounterState>(store, state => state.counter);
+const actions = new CounterActions(adapter);
+const selectors = new CounterSelectors(adapter);
+
+// Register with global store
+globalStoreServiceRegister('reduxCounter', selectors, actions);
+```
+
+### Component Usage
+
+```tsx
+// The same component works with both backends
```

Comment:
may want to call out this is to demonstrate a single interface with different backends vs expected usage